vissl/data/torchvision_dataset.py

```diff
@@ -0,0 +1,110 @@
+from typing import Tuple, List, Optional, Callable, Any
+
+from PIL import Image
+from fvcore.common.file_io import PathManager
+from torch.utils.data import Dataset
+from torchvision.datasets import CIFAR10, CIFAR100, STL10, MNIST
+
+from vissl.utils.hydra_config import AttrDict
+
+
+class TorchvisionDataset(Dataset):
+    """
+    Adapter dataset to available datasets in Torchvision.
+    The selected dataset is based on the name provided as argument.
+    This name must match the names of the dataset in torchvision.
+
+    Args:
+        cfg (AttrDict): configuration defined by user
+        data_source (string): data source ("torchvision_dataset") [not used]
+        path (string): path to the dataset
+        split (string): specify split for the dataset (either "train" or "val").
+        dataset_name (string): name of dataset.
+    """
+
+    def __init__(self,
+                 cfg: AttrDict,
+                 data_source: str,
+                 path: str,
+                 split: str,
+                 dataset_name: str):
+        super().__init__()
+        assert PathManager.isdir(path), f"Directory {path} does not exist"
+        self.dataset_name = dataset_name
+        self.path = path
+        self.split = split.lower()
+        self.dataset = self._load_dataset()
+
+    def _load_dataset(self):
+        is_train_split = self.split == "train"
+        if self.dataset_name == "CIFAR10":
+            return CIFAR10(self.path, train=is_train_split)
+        elif self.dataset_name == "CIFAR100":
+            return CIFAR100(self.path, train=is_train_split)
+        elif self.dataset_name == "STL10":
+            stl_split = "train" if is_train_split else "test"
+            return STL10(self.path, split=stl_split)
+        elif self.dataset_name == "MNIST":
+            return MNISTAdapter(root=self.path, train=is_train_split)
+        else:
+            raise ValueError(f"Unsupported dataset {self.dataset_name: str}")
```

Comment:
we should convert this to an enum following example https://github.com/facebookresearch/vissl/blob/master/vissl/models/model_helpers.py#L239-L256 :)

Reply:
I moved the names as constants in ef1f96f7d0f730ecc58cfb559f7894c38e14533f. This makes the code better and will allow us to have some documentation pointer. I am not sure the Enum would bring much in that case (the input field is a string, and should likely remain one since it is used for other formats of datasets as well, like "disk_folders").